### PR TITLE
[Semi-SL/Detection] Refactoring mpa/det/ to support multiple learning technique

### PR DESCRIPTION
**Pull request by jaegukhyun**
This PR includes refactoring mpa/det codes.

Current mpa/det's codes highly focus on class incremental learning, and this make hard to enable other training method.
To solve this issue, I split configure function in mpa/det/stage.py into two things. First is configuring configs for general fine tuning. Second is configuring configs for class-incremental learning. 
And I created ClassIncrStage, which inherit base Stage, to support class incremental learning. Now trainer at mpa/det inherits from ClassIncrStage, because class incremental learning is our default recipe.  
To enable semi-sl for detection, SemiStage has to be different with ClassIncrStage. Therefore I created SemiStage. SemiStage inherits from ClassIncrStage, and it modifies some configuration function for supporting semi-sl. Since it inherits ClassIncrStage, it can support class incremental scenarios.
As SemiStage is newly generated, Trainer and Inferencer for semi-sl is also generated, and they inherits SemiStage.

You can test this with otx branch for semi-sl(https://github.com/openvinotoolkit/training_extensions/tree/feature/otx-semisl-det). From this PR link(https://github.com/openvinotoolkit/training_extensions/pull/1398), you can find a description for this otx branch

### CODE REVIEW

File: mpa/det/incr_stage.py
Comment (supersoob) [inline mpa/det/incr_stage.py]: It would be okay to rename this to stage.py because we have seperated folder for semi.

File: mpa/det/semisl/semisl_stage.py
Comment (supersoob) [inline mpa/det/semisl/semisl_stage.py]: Same here.

File: recipes/stages/detection/finetune.py
Comment (supersoob) [inline recipes/stages/detection/finetune.py]: I'm a little worried about if this model config is proper to be inside finetune.py. If it's just overridden in semisl.py, I think it's proper to move there, but I would still wonder if it's right place to be, because recipe file is not for model cfg file. I wanna suggest to move this config to model.py inside semi/model.py in otx. If looks hard, I can agree to leave it in recipe file.

File: mpa/det/incr_stage.py
Comment (jaegukhyun) [reply mpa/det/incr_stage.py]: Incr_stage inherits base stage which is defined in stage.py Therefore it should be named different with stage.py

File: recipes/stages/detection/finetune.py
Comment (jaegukhyun) [reply recipes/stages/detection/finetune.py]: I understand what you're worrying about. Let me think about it

Comment (jaegukhyun) [reply recipes/stages/detection/finetune.py]: I moved model config to otx side. Could you check it?

**Comment by jaegukhyun** [comment]
@supersoob @sovrasov @kprokofi @JihwanEom It's time to merge this PR, please review this PR. I want you to add your implementation for semi-sl from this branch. After merging this PR, I will open OTX PR, because semi-sl cli test requires this mpa branch

File: mpa/modules/models/detectors/unbiased_teacher.py
Comment (supersoob) [inline mpa/modules/models/detectors/unbiased_teacher.py]: I want you to revise this save_dir path if you wanna remain this save_pseudo_labels methods.

Comment (jaegukhyun) [reply mpa/modules/models/detectors/unbiased_teacher.py]: Change the path to /tmp/ Thanks for review!

**Review by supersoob — APPROVED** [review]
Overall, it looks good to me. Let's wait for other's reviews. After merging this PR, I think it's gonna be good to make each PR for each task, referring this PR's code change. Thanks for your hard work, Jaeguk:)

**Review by goodsong81 — COMMENTED** [review]
I support @supersoob's point.
Why don't you create classincr directory?
Being a default option has nothing to do with class/directory structure.
Anyway, it's just a suggestion.

File: mpa/det/inferrer.py
Comment (JihwanEom) [inline mpa/det/inferrer.py:178]: Could you tell me why you removed these lines?

File: mpa/det/semisl/trainer.py
Comment (JihwanEom) [inline mpa/det/semisl/trainer.py:20]: Can we remove now?

File: mpa/modules/hooks/model_ema_hook.py
Comment (JihwanEom) [inline mpa/modules/hooks/model_ema_hook.py]: Do we need debug code?

File: mpa/modules/models/detectors/unbiased_teacher.py
Comment (JihwanEom) [inline mpa/modules/models/detectors/unbiased_teacher.py]: Same with here

File: mpa/modules/models/detectors/unbiased_teacher.py
Comment (JihwanEom) [inline mpa/modules/models/detectors/unbiased_teacher.py]: Also, looks like only for debug purpose

File: recipes/stages/detection/semisl.py
Comment (JihwanEom) [inline recipes/stages/detection/semisl.py:10]: Is False default option for detection?

Comment (jaegukhyun) [reply recipes/stages/detection/semisl.py:10]: I guess so, I copied from class-incr recipe

File: mpa/modules/hooks/model_ema_hook.py
Comment (jaegukhyun) [reply mpa/modules/hooks/model_ema_hook.py]: I'll remove this

File: mpa/det/semisl/trainer.py
Comment (jaegukhyun) [reply mpa/det/semisl/trainer.py:20]: No, It will be removed when mpa is merged with otx or Inhyuk's PR

File: mpa/det/inferrer.py
Comment (jaegukhyun) [reply mpa/det/inferrer.py:178]: I think it is unnecessary code, since model is not used or called after this code

File: mpa/modules/models/detectors/unbiased_teacher.py
Comment (jaegukhyun) [reply mpa/modules/models/detectors/unbiased_teacher.py]: I removed all debugging code.

**Review by JihwanEom — APPROVED** [review]
LGTM

**Review by supersoob — APPROVED** [review]
LGTM

**Pull request merged**
